### DISCUSSION TITLE
Embed raw binary data in JSON

### DISCUSSION THREAD

**Discussion by mmoelle1** [nlohmann/json#4011] [Q&A]
I would like to embed externally serialised binary data (of type `torch::serialize::OutputArchive`) into a JSON object so that I can store it to file and create a `torch::serialize::InputArchive` object upon loading the JSON file. Any help is appreciated.

**Comment by nlohmann** [comment]
Plain JSON text has no native binary type, so raw bytes can't be embedded directly — you need to encode them as a string first, typically with base64 (the library doesn't ship a base64 encoder/decoder, so you'd bring your own, e.g. from a small header-only library). Alternatively, if you don't need the output to be human-readable JSON text, switch to one of the library's supported binary formats (CBOR, MessagePack, BSON, UBJSON, or BJData), all of which have a native json::binary_t you can assign a std::vector<uint8_t> to directly without any text encoding overhead — see docs/mkdocs/docs/features/binary_formats/.

---
*This reply was drafted by [Claude Code](https://claude.com/claude-code) on behalf of @nlohmann, as part of a review of open discussions.*
